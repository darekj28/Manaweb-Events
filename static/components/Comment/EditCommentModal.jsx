var React = require('react');
export default class EditCommentModal extends React.Component {
	constructor(props) {
		super(props);
		this.state = {commentContent : this.props.comment.commentContent};
	}
	handleCommentEditChange(e) {
		this.setState({ commentContent : e.target.value});
	}
	handleCommentEditSubmit() {
		var obj = {unique_id : this.props.comment.unique_id,
						field_name : "body", 
						field_data : this.state.commentContent};

		$.ajax({
			type : 'POST',
			url  : '/editComment',
			data : JSON.stringify(obj, null, '\t'),
			contentType: 'application/json;charset=UTF-8'
		});
		this.props.handleCommentEdit(this.props.comment, this.state.commentContent);
	}
	componentWillReceiveProps(nextProps) {
		this.setState({ commentContent : nextProps.comment.commentContent });
	}
	render() {
		return(
			<div className="modal fade" id="EditCommentModal" aria-hidden="true">
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-label="close">
								<span aria-hidden="true">&times;</span>
							</button>
							<div className="modal-title">Edit Comment</div>
						</div>
						<div className="modal-body" id="EditCommentModalBody">
							<textarea className="form-control edit-post" id="FeedCommentBody" 
									value={this.state.commentContent} 
									onChange={this.handleCommentEditChange.bind(this)}></textarea>
						</div>
						<div className="modal-footer">
							<button id="ecm_submit" type="button" className="btn post-button" data-dismiss="modal" 
									onClick={this.handleCommentEditSubmit.bind(this)}>Submit</button>
						</div>
					</div>
				</div>
			</div>
		);
	}
}