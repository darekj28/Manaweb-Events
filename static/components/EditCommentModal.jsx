var React = require('react');
var $ = require('jquery');
export default class EditCommentModal extends React.Component {
	constructor(props) {
		super(props);
		this.state = {commentContent : this.props.comment.commentContent};
		this.handleCommentEditChange = this.handleCommentEditChange.bind(this);
		this.handleCommentEditSubmit = this.handleCommentEditSubmit.bind(this);
	}
	handleCommentEditChange(e) {
		this.setState({ commentContent : e.target.value});
	}
	handleCommentEditSubmit() {
		var obj = {unique_id : this.props.comment.unique_id,
						field_name : "body", 
						field_data : this.state.commentContent};

		$.ajax({
			type : 'POST',
			url  : '/editComment',
			data : JSON.stringify(obj, null, '\t'),
			contentType: 'application/json;charset=UTF-8'
		});
		this.props.refreshFeed();
	}
	componentWillReceiveProps(nextProps) {
		this.setState({ commentContent : nextProps.comment.commentContent });
	}
	render() {
		return(
			<div className="modal fade" id="EditCommentModal" aria-hidden="true">
				<div className="modal-dialog" role="document">
					<div className="modal-content">
						<div className="modal-header">
							<button type="button" className="close" data-dismiss="modal" aria-label="close">
								<span aria-hidden="true">&times;</span>
							</button>
							<h4 className="modal-title">Edit Comment</h4>
						</div>
						<div className="modal-body" id="EditCommentModalBody">
							<textarea className="form-control EditFeedCommentBody" id="FeedCommentBody" 
									value={this.state.commentContent} 
									onChange={this.handleCommentEditChange}></textarea>
						</div>
						<div className="modal-footer">
							<button type="button" className="btn btn-default" data-dismiss="modal" 
									onClick={this.handleCommentEditSubmit}>Submit</button>
						</div>
					</div>
				</div>
			</div>
		);
	}
}