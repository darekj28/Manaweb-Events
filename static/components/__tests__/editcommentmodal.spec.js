import EditCommentModal from '../EditCommentModal.jsx';

describe('EditCommentModal', () => {
	it('should call handleCommentEditSubmit when submit is clicked', () => {
		window.$ = {ajax: jest.fn()}
		const refreshFeed = jest.fn();
		const comment = {unique_id : 1};
		const wrapper = mount(
			<EditCommentModal comment={comment} refreshFeed={refreshFeed}/>
			);
		const btn = wrapper.find('#ecm_submit');
		btn.simulate('click');
		expect(refreshFeed).toHaveBeenCalled();
	});
	it('renders correctly', () => {
		const refreshFeed = jest.fn();
		const comment = {unique_id : 1};
		const ecm = renderer.create(
			<EditCommentModal comment={comment} refreshFeed={refreshFeed}/>
			);
		expect(ecm).toMatchSnapshot();
	});

})