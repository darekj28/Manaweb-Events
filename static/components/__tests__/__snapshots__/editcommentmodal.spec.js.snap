exports[`EditCommentModal renders correctly 1`] = `
<div
  aria-hidden="true"
  className="modal fade"
  id="EditCommentModal">
  <div
    className="modal-dialog"
    role="document">
    <div
      className="modal-content">
      <div
        className="modal-header">
        <button
          aria-label="close"
          className="close"
          data-dismiss="modal"
          type="button">
          <span
            aria-hidden="true">
            Ã—
          </span>
        </button>
        <h4
          className="modal-title">
          Edit Comment
        </h4>
      </div>
      <div
        className="modal-body"
        id="EditCommentModalBody">
        <textarea
          className="form-control EditFeedCommentBody"
          id="FeedCommentBody"
          onChange={[Function]}
          value={undefined} />
      </div>
      <div
        className="modal-footer">
        <button
          className="btn btn-default"
          data-dismiss="modal"
          id="ecm_submit"
          onClick={[Function]}
          type="button">
          Submit
        </button>
      </div>
    </div>
  </div>
</div>
`;
